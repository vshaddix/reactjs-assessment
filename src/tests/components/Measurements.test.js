import React from 'react';
import { create } from "react-test-renderer";
import Measurement from '../../components/Measurements/Measurements';

it('renders (snapshot test)', () => {
  const component = create(<Measurement />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when passed full data', () => {
  const measurements = [
    {
      parameter: "test",
      value: "1",
      unit: 7
    },
    {
      parameter: "test2",
      value: "2",
      unit: 2
    }
  ];

  const component = create(<Measurement measurements={measurements} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when no value is passed for only one measure', () => {
  const measurements = [
    {
      parameter: "test",
      value: "1",
      unit: 7
    },
    {
      parameter: "test2",
      unit: 2
    }
  ];

  const component = create(<Measurement measurements={measurements} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when empty array is passed as measurements', () => {
  const measurements = [];

  const component = create(<Measurement measurements={measurements} />);

  expect(component.toJSON()).toMatchSnapshot();
});
