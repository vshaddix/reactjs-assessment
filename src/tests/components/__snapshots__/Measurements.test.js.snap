// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `
<div
  className="ui placeholder"
>
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
</div>
`;

exports[`renders the component correctly when empty array is passed as measurements 1`] = `
<div
  className="event"
>
  <div
    className="content"
  >
    <div
      className="summary"
    />
  </div>
</div>
`;

exports[`renders the component correctly when no value is passed for only one measure 1`] = `
<div
  className="event"
>
  <div
    className="content"
  >
    <div
      className="summary"
    >
      <div>
        <span>
          test
        </span>
        : 
        <span>
          1
           
          7
        </span>
      </div>
      <div>
        <span>
          test2
        </span>
        : 
        <span>
          -
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component correctly when passed full data 1`] = `
<div
  className="event"
>
  <div
    className="content"
  >
    <div
      className="summary"
    >
      <div>
        <span>
          test
        </span>
        : 
        <span>
          1
           
          7
        </span>
      </div>
      <div>
        <span>
          test2
        </span>
        : 
        <span>
          2
           
          2
        </span>
      </div>
    </div>
  </div>
</div>
`;
